gems/pending/util/MiqSockUtil.rb

```diff
@@ -8,6 +8,9 @@ def self.getHostName
   # Return the cannonical name for host in DNS
   def self.getFullyQualifiedDomainName
     Socket.gethostbyname(Socket.gethostname).first
+  rescue
+    $log.error "The appliance hostname could not be resolved. Please make sure to set a valid fully qualified hostname."
```

Comment:
Honestly it should be caught where you do SSA of OpenStack host.  That line you guy is also a generic utility and doesn't make sense to rescue there.